### PR TITLE
introduce a new "bad-setting" unit load state in order to improve "systemctl status" output when bad settings are used

### LINKED ISSUE #9107: WorkingDirectory ending in "/." leads to "Operation not permitted" error on load
**systemd version the issue has been seen with**
> 237

**Used distribution**
 > ubuntu 18.04
> Linux web 4.15.0-12-generic #13-Ubuntu SMP Thu Mar 8 06:24:47 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

**Expected behaviour you didn't see**
 > unit file should load

**Unexpected behaviour you saw**
 > unit file fails to load 

**Steps to reproduce the problem**

- create simple unit file  `/etc/systemd/system/bugdemo.service`

 > [Unit]
 > Description=demonstrate possible bug
 >
 > [Service]
 > ExecStart=/bin/cat
 > WorkingDirectory=/etc

- reload, check status
   
 > $ systemctl daemon-reload
 > $ systemctl status bugdemo.service
 >   ● bugdemo.service - demonstrate possible bug
 >    Loaded: loaded (/etc/systemd/system/bugdemo.service; static; vendor preset: enabled)

- add `/.` to WorkingDirectory

 > [Unit]
 > Description=demonstrate possible bug
 >
 > [Service]
 > ExecStart=/bin/cat
 > WorkingDirectory=/etc/.

- reload, check status

 > $ systemctl daemon-reload
 > $ systemctl status bugdemo.service
 >   ● bugdemo.service - demonstrate possible bug
 >    Loaded: error (Reason: Operation not permitted)

Even if that's intended behaviour, the error message is not very helpful.

Background: I'm creating the unit-file with ansible and there's an optional extra part to the working directory. That part default's to `.`, so I can combine the base path with the optional extra bit, even if the extra part is not used. I can work around this with a conditional, but it seems odd to me that systemd refuses to load the unit-file with the extra `/.`.

### PR DESCRIPTION
**Pull request by poettering**
An attempt to fix #9107. It's orthogonal to @yuwata's work on #9149. This PR is an attempt to improve "systemctl status" output when we encounter an error loading some unit file due to a bad setting. @yuwata's work is about not making WorkingDirectory= on non-normalized paths fail in the first place. Both PRs should be considered independently of each other.

With this PR, systemctl output for a unit with a bad WorkingDirectory= setting will now look like this:

```
# systemctl status foo.service
● foo.service
   Loaded: bad-setting (Reason: Unit foo.service has a bad unit file setting.)
   Active: inactive (dead)

Jun 01 18:15:09 image systemd[1]: /etc/systemd/system/foo.service:3: Working directory path 'kkkk' is not absolute.
```

### CODE REVIEW

File: src/basic/unit-def.h [hunk 1]
```diff
@@ -30,8 +30,9 @@ typedef enum UnitType {
 typedef enum UnitLoadState {
         UNIT_STUB = 0,
         UNIT_LOADED,
-        UNIT_NOT_FOUND,
-        UNIT_ERROR,
+        UNIT_NOT_FOUND,    /* error condition #1: unit file not found */
+        UNIT_BAD_SETTING,  /* error condition #2: we couldn't parse some essential unit file setting */
+        UNIT_ERROR,        /* error condition #3: other "system" error, catchall for the rest */
```
Comment (yuwata) [inline src/basic/unit-def.h]: missing space in the comment, 'catchall' -> 'catch all',

Comment (filbranden) [reply src/basic/unit-def.h]: 'catchall' is actually [correct](https://www.google.com/search?q=catchall).

Comment (yuwata) [reply src/basic/unit-def.h]: Thanks.

**Comment by poettering** [comment]
I force pushed a new version now, that makes this mergable against master again.

**Comment by keszybz** [comment]
LGTM.

From #9107:
```
● workingdir.service - demonstrate possible bug
   Loaded: bad-setting (Reason: Unit workingdir.service has a bad unit file setting.)
   Active: inactive (dead)

Jun 11 14:28:30 krowka systemd[1]: Started demonstrate possible bug.
Jun 11 14:28:59 krowka systemd[1]: /etc/systemd/system/workingdir.service:6: = path is not normalizedWorkingDirecto
```

I'll push a separate PR to fix the error message.

**Pull request merged**